addon/base-record-array.js

```diff
@@ -24,6 +24,43 @@ import { recordIdentifierFor } from '@ember-data/store';
  * @class BaseRecordArray
  */
 let BaseRecordArray;
+let BaseRecordArrayProxyHandler;
+
+if (PROXY_MODEL_CLASS) {
+  const convertToInt = (prop) => {
+    if (typeof prop === 'symbol') return null;
+
+    const num = Number(prop);
+
+    if (isNaN(num)) return null;
+
+    return num % 1 === 0 ? num : null;
+  };
```

Comment:
should this be an arrow function?  we don't really need the binding do we?

isn't `function(prop) { /* ... */ }` more appropriate?